INSS.EIIR.DailyExtract/SyncData.cs

```diff
@@ -27,20 +27,20 @@ public SyncData(ILogger<SyncData> logger, IResponseUseCase<SyncDataResponse> res
 
         }
 
-        [Function("SyncData")]
-        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "get", "post")] HttpRequest req)
+        [Function(nameof(SyncData))]
+        public async Task Run([ActivityTrigger] string name)
         {
             _logger.LogInformation("SyncData started");
 
             var response = await _responseUseCase.Handle();
 
             if (response.IsError)
             {
-                return new BadRequestObjectResult($"SyncData failed with {response.ErrorCount} errors. Message: {response.ErrorMessage}");
+                _logger.LogWarning($"SyncData failed with {response.ErrorCount} errors. Message: {response.ErrorMessage}");
```

Comment:
return statement helps if the function fails then it will show in the RED in Azure Portal.
replacing with log , shows always green in Azure Portal even there is a failure.
why are we changing previous implementation?

Reply:
The reason for the change is to run SyncData as a Durable Function due to the time it was taking to process.  Originally called as a http request Azure was giving up as it was exceeding various timeouts and logging an error - even when it SyncData completed successfully.  The change was to allow the process to complete elegantly... by this stage nothing is listening for a response.